src/Controller/UserController.php

```diff
@@ -19,10 +19,16 @@ class UserController extends AbstractController
      */
     public function show(User $user, EventRepository $eventRepository): Response
     {
-        $events = $eventRepository->findBy(['user' => $user]);
-        return $this->render('user/show.html.twig', [
-            'user' => $user,
-            'events' => $events
-        ]);
+        $options = ['user' => $user];
+        $view = 'user/';
+        if (in_array(User::ROLE_AMBASSADOR, $user->getRoles())) {
+            $events = $eventRepository->findBy(['user' => $user]);
+            $view .= 'show_ambassador.html.twig';
+            $options['events'] = $events;
+        } elseif (in_array(User::ROLE_CREATOR, $user->getRoles())) {
+            $view .= 'show_creator.html.twig';
+        }
+
+        return $this->render($view, $options);
```

Comment:
vu que l'$option est facultative, je trouve ça plus propre de faire deux return $this->render, c'est plus clair je trouve. J'aime pas trop le fait de mettre le nom du template dans une variable.
De plus, tu peux optimiser le check des roles avec $this->isGranted()  à vérifier si on peut passer un objet $user en second paramètre (je pense que oui mais c'est pas clair sur la doc) ou si c'est uniquement réservé à l'user actuellement connecté.
https://symfony.com/doc/current/security.html#hierarchical-roles


petite optimisation
```
    public function show(User $user, EventRepository $eventRepository): Response
    {
         if ($this->isGranted('ROLE_AMBASSADOR,$user->getRoles())) { 
             $events = $eventRepository->findByUser($user); 
            return $this->render('user/show_ambassador.html.twig', ['events'=>$events]); 
         } elseif ($this->isGranted('ROLE_CREATOR',$user->getRoles())) { 
            return $this->render('user/show_creator.html.twig'); 
        }
}
```